### DISCUSSION TITLE
How to set up specific c++ standard for ALE?

### DISCUSSION THREAD

**Discussion by jacklanda** [w0rp/ale#3905] [Q&A - Ask for help with problems]
ALE will sometimes give the error lint where Some c++ new features could only be accessed when we set the complie flag (e.g `-std==c++17`), however, these error lints should not appear in fact.

Setting the c++ standard(e.g. c++14, c++17) for ALE is a bit confused, I would like to figure it out and so search many issues and contents in doc,  but no help with this problem.

Is there anybody know how to solve this issue (maybe need to add a configuration in `.vimrc` ?

**Comment by matrdr** [comment]
I fixed it for the linter by using `let g:ale_cpp_cc_options = '-std=c++17 -Wall'` in my `.vimrc`.

After I installed the _ccls_ language server for C(++), the error reappeared, so I had to configure the language server too by further adding
```vim
let g:ale_cpp_ccls_init_options =
    \{
    \       'clang': {
    \               'extraArgs': ['-std=c++17',],
    \       },
    \}
```

If you use a different language server, the configuration will be different of course.
